litestar/contrib/minijinja.py

```diff
@@ -105,7 +105,6 @@ def __init__(self, directory: Path | list[Path] | None = None, engine_instance:
             directory: Direct path or list of directory paths from which to serve templates.
             engine_instance: A Minijinja Environment instance.
         """
-        super().__init__(directory, engine_instance)
```

Comment:
Why are these removed? Iiuc this could cause issues when using these classes in multiple inheritance

Reply:
Mypy says it's unsafe: https://mypy.readthedocs.io/en/stable/error_code_list.html#check-that-call-to-an-abstract-method-via-super-is-valid-safe-super

> Iiuc this could cause issues when using these classes in multiple inheritance

Can you demo this for me please?

Reply:
Wouldn't the protocol `__init__` methods need to call super for that scenario too?

Reply:
I *think* these only happen when they are used as "mixins" in which case I would assume the protocol has to delegate to the super  `__init__`, we can maybe mark is as `final` indicating this should not be inherited?

Reply:
Shouldn't `MiniJinjaTemplateEngine` inherit from the protocol in this scenario?

Reply:
thank you, edited, so

>  the protocol __init__ methods need to call super for that scenario too?
seems like it should

Reply:
> Wouldn't the protocol `__init__` methods need to call super for that scenario too?

Ah yeah, good point. Uh. I guess the question is how multiple inheritance works with protocols? Actually I'm not sure how they behave at runtime.

Reply:
Okay, I checked. They don't seem to call `super().__init__`, so we'd have to add that explicitly to make them work with multi-inheritance anyway. In that case I'm fine with this change as-is.